runtime/lua/vim/lsp/rpc.lua

```diff
@@ -377,6 +378,23 @@ local function create_and_start_client(cmd, cmd_args, handlers, extra_spawn_para
       decoded.error = convert_NIL(decoded.error)
       decoded.result = convert_NIL(decoded.result)
 
+      -- For RequestCancelled, we want to ignore the handlers since this is an
+      -- RPC internal handling code and we shouldn't report this to users.
```

Comment:
nit (but also in general): less-verbose wording:

    -- Do not surface RequestCancelled to users, it is RPC-internal.